kratos/python_scripts/response_functions/response_function_interface.py

```diff
@@ -44,5 +33,8 @@ def Finalize(self):
     def GetValue(self):
         raise NotImplementedError("GetValue needs to be implemented by the derived class")
 
-    def GetShapeGradient(self):
-        raise NotImplementedError("GetShapeGradient needs to be implemented by the derived class")
+    def GetNodalGradient(self, variable):
+        raise NotImplementedError("GetNodalGradient needs to be implemented by the derived class")
+
+    def GetElementalGradient(self, variable):
+        raise NotImplementedError("GetElementalGradient needs to be implemented by the derived class")
```

Comment:
To be complete we should also add `GetConditionalGradient`.

Reply:
added it.